### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by juanavelez**
Some minor refactoring to remove duplicate code. Had to create static class HandlerUtil because complaints from codegen about static methods in interface.

Signed-off-by: Juan A. Velez <jvelez@chibchasoft.com>

### CODE REVIEW

File: src/main/java/io/vertx/ext/sql/util/HandlerUtil.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import io.vertx.ext.sql.SQLConnection;
+
+import java.util.List;
+
+/**
+ * Collection of static methods that deal with {@link Handler}s within the Context of a SQLConnection or related SQL-Objects
+ *
+ * @author <a href="mailto:jvelez@chibchasoft.com">Juan VELEZ</a>
+ */
+public final class HandlerUtil {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/util/HandlerUtil.java]: make the method package private

File: src/main/java/io/vertx/ext/sql/util/HandlerUtil.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  /**
+   * Returns a {@link Handler} for {@link AsyncResult} of {@link ResultSet} that will delegate the outcome of the result
+   * (only a single row) to the passed handler. If the {@code AsyncResult} is failed then the failure will be propagated
+   * to the delegate. If the {@code ResultSet} is null or empty, a null will be the result passed to the delegate handler
+   * @param handler the target handler of the result
+   * @return the new handler
+   */
+  public static Handler<AsyncResult<ResultSet>> handleResultSetSingleRow(Handler<AsyncResult<@Nullable JsonArray>> handler) {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/util/HandlerUtil.java]: make the method package private

File: src/main/java/io/vertx/ext/sql/util/HandlerUtil.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  /**
+   * Returns a {@link Handler} for {@link AsyncResult} of a {@code SQL} result that regardless of the outcome will close
+   * the passed {@link SQLConnection}. If the {@code AsyncResult} is failed then the failure will be propagated
+   * to the delegate handler. If it was successful, the result will be propagated to the delegate handler. In any case
+   * the delegation to the handler will be done as part of closing the {@code connection}.
+   * @param conn the connection to close
+   * @param handler the target handler of the result
+   * @return the new handler
+   */
+  public static <T> Handler<AsyncResult<T>> closeAndHandleResult(SQLConnection conn, Handler<AsyncResult<T>> handler) {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/util/HandlerUtil.java]: make the method package private

**Review by vietj — CHANGES_REQUESTED** [review]
some changes required.

**Comment by juanavelez** [comment]
@vietj I have made the requested changes. Please review. Thank you

**Review by vietj — APPROVED** [review]

**Pull request merged**